### PR TITLE
Configurable views

### PR DESCRIPTION
**Pull request by jimCresswell**
Provide configured views in a `specs.json` file and use them to alter the UI of the project page. Where they exist, views can be chosen from a select element.

See individual commit messages for details.

### CODE REVIEW

**Comment by sponte** [comment]
@jimCresswell thanks, it's pretty cool to see it working. Could we add an option for a default view in the specs.json? Also, could we rename the file to .specs or .specs.json - this would make it hidden in most file browsers.

**Comment by jimCresswell** [comment]
I think defaults are a great idea.

I'd rather keep the configuration file not hidden, similar to bower.json or package.json. What advantage do you see in the file being hidden in file browsers?

**Comment by jimCresswell** [comment]
@sponte as discussed, default views implemented and the `specs.json` file can optionally be named `.specs.json`.

File: README.md [hunk 1]
```diff
@@ -37,11 +37,21 @@ To install the latest version from npm
...
 * SPECS_OUT_DIR - Specifies location of project-data folder, by default this lives in the root of the project.
 * SPECS_ALLOW_INSECURE_SSL - Disables SSL certificate verification when cloning repos.
-* SPECS_EXCLUDED_PATHS - Excludes certain paths (globally) from projects.
-* SPECS_PATHS_TO_HIDE - Specify beginnings of directory paths to hide on the project page. E.g. if you specified 'src/test/specs' then a directory 'src/test/specs/my_wondeful_feature' would be displayed in the directory list as 'my_wonderful_feature'. This only affects display logic, not the underlying data.
+
+### Project specific configuration via a specs.json file
+The file should be in the root of your repository. The file an be named `specs.json` or `.specs.json` .See the [.specs.json](.specs.json) file in the root of this repository for an example.
+
+Keys:
+ * views: An object with keys specifying named view configurations. Allowed keys are:
```
Comment (sponte) [inline README.md]: can we add a line for default

Comment (jimCresswell) [inline README.md]: Done.

**Pull request merged**